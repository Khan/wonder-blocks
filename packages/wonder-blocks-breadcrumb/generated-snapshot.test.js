// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-breadcrumb
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");
import Breadcrumb from "./components/breadcrumb.js";
import BreadcrumbItem from "./components/breadcrumb-item.js";

describe("wonder-blocks-breadcrumb", () => {
    it("example 1", () => {
        const Link = require("@khanacademy/wonder-blocks-link").default;
        const {
            BreadcrumbItem,
        } = require("@khanacademy/wonder-blocks-breadcrumb");

        const example = (
            <Breadcrumb>
                <BreadcrumbItem>
                    <Link href="https://khanacademy.org/">Home</Link>
                </BreadcrumbItem>
                <BreadcrumbItem aria-current="page">
                    Current page
                </BreadcrumbItem>
            </Breadcrumb>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const Link = require("@khanacademy/wonder-blocks-link").default;
        const {
            BreadcrumbItem,
        } = require("@khanacademy/wonder-blocks-breadcrumb");

        const example = (
            <BreadcrumbItem>
                <Link href="https://khanacademy.org/">Home</Link>
            </BreadcrumbItem>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 3", () => {
        const Link = require("@khanacademy/wonder-blocks-link").default;
        const {
            BreadcrumbItem,
        } = require("@khanacademy/wonder-blocks-breadcrumb");

        const example = <BreadcrumbItem>Current page</BreadcrumbItem>;
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
