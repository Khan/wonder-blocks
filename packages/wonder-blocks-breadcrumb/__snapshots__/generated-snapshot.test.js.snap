// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-breadcrumb example 1 1`] = `
<nav
  aria-label="Breadcrumb"
>
  <ol
    className=""
    style={
      Object {
        "listStyle": "none",
        "margin": 0,
        "overflow": "hidden",
        "padding": 0,
      }
    }
  >
    <li
      className=""
      style={
        Object {
          "::after": Object {
            "content": "'∙'",
            "display": "inline-block",
            "margin": "0 6",
          },
          ":lastChild": Object {
            "::after": Object {
              "content": "''",
            },
          },
          "display": "inline-block",
          "margin": "0 6",
        }
      }
    >
      <a
        className=""
        href="https://khanacademy.org/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragStart={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "color": "#1865f2",
            "cursor": "pointer",
            "outline": "none",
            "textDecoration": "none",
          }
        }
        tabIndex={0}
      >
        Home
      </a>
    </li>
    <li
      className=""
      style={
        Object {
          "::after": Object {
            "content": "'∙'",
            "display": "inline-block",
            "margin": "0 6",
          },
          ":lastChild": Object {
            "::after": Object {
              "content": "''",
            },
          },
          "display": "inline-block",
          "margin": "0 6",
        }
      }
    >
      Current page
    </li>
  </ol>
</nav>
`;

exports[`wonder-blocks-breadcrumb example 2 1`] = `
<li
  className=""
  style={
    Object {
      "::after": Object {
        "content": "'∙'",
        "display": "inline-block",
        "margin": "0 6",
      },
      ":lastChild": Object {
        "::after": Object {
          "content": "''",
        },
      },
      "display": "inline-block",
      "margin": "0 6",
    }
  }
>
  <a
    className=""
    href="https://khanacademy.org/"
    onBlur={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "color": "#1865f2",
        "cursor": "pointer",
        "outline": "none",
        "textDecoration": "none",
      }
    }
    tabIndex={0}
  >
    Home
  </a>
</li>
`;

exports[`wonder-blocks-breadcrumb example 3 1`] = `
<li
  className=""
  style={
    Object {
      "::after": Object {
        "content": "'∙'",
        "display": "inline-block",
        "margin": "0 6",
      },
      ":lastChild": Object {
        "::after": Object {
          "content": "''",
        },
      },
      "display": "inline-block",
      "margin": "0 6",
    }
  }
>
  Current page
</li>
`;
